File: Easy/src/Java/J136/SingleNumber.java
package Java.J136;

public class SingleNumber {

    public int singleNumber(int[] nums) {
        int result = 0;
        for (int num : nums) {
            System.out.println("Before XOR: result = " + result + ", num = " + num);
            result ^= num;
            System.out.println("After XOR: result = " + result);
        }
        return result;
    }

    public static void main(String[] args) {
        SingleNumber sn = new SingleNumber();
        int[] nums = {4,1,2,1,2};
        int res = sn.singleNumber(nums);
        System.out.println("Single number is: " + res);
    }
}

File: Easy/src/Go/Single Number.go
package main

import "fmt"

func singleNumber(nums []int) int {
	res := 0
	for i := 0; i < len(nums); i++ {
		res ^= nums[i]
	}
	return res
}

func main() {
	// Примеры использования
	nums1 := []int{2, 2, 1}
	nums2 := []int{4, 1, 2, 1, 2}
	nums3 := []int{1}

	fmt.Println("Пример 1:", singleNumber(nums1))
	fmt.Println("Пример 2:", singleNumber(nums2))
	fmt.Println("Пример 3:", singleNumber(nums3))
}
